### ISSUE TITLE
Model.tag_names or similar functionality?

### ISSUE THREAD

**Issue by guaguasi** [pat/gutentag#61]
I just recently found your gem and I can't seem to find a cleaner way to get a list of all tags for a particular model. I'm hoping for something along the lines of the following:

`Article.tag_names #=> ['business', 'leisure', 'nsfw']`

At the moment, I'm using the following. It feels a bit hacky with the multi-tenancy. Is this the best option?

```
module Taggable
  extend ActiveSupport::Concern

  included do
    Gutentag::ActiveRecord.call self
  end

  class_methods do
    def tag_names
      Gutentag::Tag.joins(:taggings)
        .where(gutentag_taggings: { taggable_type: self, taggable_id: pluck(:id) })
        .distinct.pluck(:name)
    end
  end
end
```

Thanks!

**Comment by pat** [comment]
There's nothing inbuilt in Gutentag to do what you're after… but I can see it would indeed be useful. I'll look at putting something proper in (and quite likely through the code you've suggested!)

**Comment by pat** [comment]
Okay, so what I've pushed to master is the `Gutentag::Tag.names_for_scope` method, which accepts a single parameter, which can be a model or a scope/relation. I deliberately avoided putting this method on the taggable model, as I would much prefer avoiding overloading more methods into models outside of the gem itself.

If it's a model, then it will return all tag names used by instances of that model. If it's a relation, it'll filter the tags by only records within that scope. This all occurs with a single SQL query.

```ruby
Gutentag::Tag.names_for_scope(Article)
Gutentag::Tag.names_for_scope(Article.where(:created_at => 1.week.ago..1.second.ago))
```

@guaguasi you did mention multi-tenancy, which I presume means the model may be scoped by default? I believe this should behave appropriately, but let me know if it doesn't!

**Comment by pat** [comment]
Oh, and it's worth noting here as much as anywhere: introducing this feature means I'll officially drop support for Rails 3.2 and MRI 2.2 when it becomes part of a proper gem release. Given Rails 6 will be out soon, I feel like supporting 4.x-6.x and MRI 2.3-2.6 is a fair target. The code may indeed work on MRI 2.2 - and everything but this feature will work on Rails 3.2 - but I won't be testing against those versions moving forward.

**Comment by pat** [comment]
Just released v2.5.0 with this feature.
